pkg/ingester/ingester.go

```diff
@@ -405,14 +405,8 @@ func New(cfg Config, limits *validation.Overrides, ingestersRing ring.ReadRing,
 		i.ownedSeriesService = newOwnedSeriesService(i.cfg.OwnedSeriesUpdateInterval, i.lifecycler.ID, ingestersRing, log.With(i.logger, "component", "owned series"), registerer, i.limits.IngestionTenantShardSize, i.getTSDBUsers, i.getTSDB)
 	}
 
-	// Init the limter and instantiate the user states which depend on it
-	i.limiter = NewLimiter(
-		limits,
-		ingestersRing,
-		cfg.IngesterRing.ReplicationFactor,
-		cfg.IngesterRing.ZoneAwarenessEnabled,
-		cfg.IngesterRing.InstanceZone,
-	)
+	var limiterStrategy limiterRingStrategy
+	limiterStrategy = newIngesterRingLimiterStrategy(ingestersRing, cfg.IngesterRing.ReplicationFactor, cfg.IngesterRing.ZoneAwarenessEnabled, cfg.IngesterRing.InstanceZone, i.limits.IngestionTenantShardSize)
```

Comment:
[nit] To make the intention more clear, I would move `newIngesterRingLimiterStrategy` in a `else` branch of `if ingestCfg := cfg.IngestStorageConfig; ingestCfg.Enabled {` so that we don't initialise the "wrong" and then override it when ingest storage is enabled.

Reply:
Done in https://github.com/grafana/mimir/pull/7424/commits/9811e539683e5ab52d353d1e3d0533992561ed60.